### ISSUE TITLE
Python 3.5 - missing libpython3.5m.so (shared library)

### ISSUE THREAD

**Issue by matysek** [travis-ci/travis-ci#4834]
Python 3.5 environment in travis is missing shared python library libpython3.5m.so.

This is the debug output for a test job (https://travis-ci.org/matysek/pyinstaller/builds/81371963)
```
$ python --version
Python 3.5.0
$ pip --version
pip 7.1.2 from /home/travis/virtualenv/python3.5.0/lib/python3.5/site-packages (python 3.5)
before_install.1
0.01s$ ls /opt
pyenv  python
before_install.2
0.01s$ ls /opt/python/
2.6.9  2.7.9  3.2.5  3.3.5  3.4.2  3.5.0  pypy-2.5.0  pypy3-2.4.0
before_install.3
0.01s$ ls /opt/python/2.7.9/
bin  include  lib  share
before_install.4
0.01s$ ls /opt/python/2.7.9/lib
libpython2.7.so  libpython2.7.so.1.0  pkgconfig  python2.7
before_install.5
0.01s$ ls /opt/python/3.3.5/
bin  include  lib  share
before_install.6
0.01s$ ls /opt/python/3.3.5/lib
libpython3.3m.so  libpython3.3m.so.1.0	libpython3.so  pkgconfig  python3.3
before_install.7
0.01s$ ls /opt/python/3.4.2/
bin  include  lib  share
before_install.8
0.01s$ ls /opt/python/3.4.2/lib
libpython3.4m.so  libpython3.4m.so.1.0	libpython3.so  pkgconfig  python3.4
before_install.9
0.01s$ ls /opt/python/3.5.0/
bin  include  lib  share
before_install.10
0.01s$ ls /opt/python/3.5.0/lib
libpython3.5m.a  pkgconfig  python3.5
```

Our project PyInstaller requires shared python library. Running tests with other Python version does not complain about about missing libpython: https://travis-ci.org/pyinstaller/pyinstaller/jobs/81375212

**Comment by BanzaiMan** [comment]
Thanks for opening the issue.

This is quite curious. We use python-build to compile from source (https://github.com/travis-ci/cpython-builder/blob/version/build-python#L17-L18), very much the same way others were built.

I am not sure how the difference came about.

**Comment by matysek** [comment]
This issue might be related: https://github.com/yyuu/pyenv/issues/443

**Comment by BanzaiMan** [comment]
That issue is about OS X, no?

**Comment by matysek** [comment]
It is but it gives instructions how to build Python with pyenv and using option `--enable-shared`: https://github.com/yyuu/pyenv/wiki/Home/_compare/45570ea%5E...45570ea

Sorry, I was not clear enough.

**Comment by BanzaiMan** [comment]
Ah, I see! Thanks for that link. I've added the flag to our build process and rebuilt the package.

https://travis-ci.org/BanzaiMan/travis_production_test/builds/81587214#L101 it now has shared objects instead of the archive.

**Comment by matysek** [comment]
@BanzaiMan Thanks, it fixes PyInstaller tests with Python 3.5: https://travis-ci.org/pyinstaller/pyinstaller/jobs/81581825

**Comment by BanzaiMan** [comment]
Super. I'm closing this now.
